src/App.tsx

```diff
@@ -1,61 +1,109 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+
+  const [userIdError, setUserIdError] = useState(false);
+  const [inputTitleError, setInputTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [inputTitle, setInputTitle] = useState('');
+
+  const resetFunc = () => {
+    setUserId(0);
+    setInputTitle('');
+  };
+
+  const handleInputChange = (newInput: string) => {
+    setInputTitleError(false);
+    setInputTitle(newInput);
+  };
+
+  const handleUserChange = (newUser: number) => {
+    setUserIdError(false);
+    setUserId(newUser);
+  };
+
+  const handleAddClick = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+
+    if (!userId) {
+      setUserIdError(true);
+    }
+
+    if (!inputTitle.length) {
+      setInputTitleError(true);
+    }
+
+    if (!inputTitle.length || !userId) {
+      return;
+    }
+
+    const newTodo = {
+      id: [...todos]
+        .sort((todo1, todo2) => todo2.id - todo1.id)[0].id + 1,
```

Comment:
you can just take the id of the last todo in array and increase it +1 (no need to sort)